### PR TITLE
ISPN-14114 Fix TLS 1.3 with cipher suites

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14114

Based on https://github.com/infinispan/infinispan/pull/10281

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/proc_server_configuring_ssl_engine.adoc [hunk 1]
```diff
@@ -25,7 +25,7 @@ Omitting the `enabled-protocols` attribute allows any TLS version.
 ----
 ====
 +
-. Configure {brandname} to use one or more cipher suites with the `enabled-ciphersuites` attribute.
+. Configure {brandname} to use one or more cipher suites with the `enabled-ciphersuites-filter` attribute (for TLSv1.2 and below) and the `enabled-ciphersuites-names` (for TLSv1.3).
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_server_configuring_ssl_engine.adoc:28]: I'm no TLS expert, but are the terms "filter" and "names" synonymous with `TLS < 1.3` and `1.3` respectively?

To me keeping `enabled-ciphersuites` for v1.1 and v1.2 and adding `enabled-ciphersuites-1-3` for v1.3 is much more readable and self-explanatory.

Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/proc_server_configuring_ssl_engine.adoc:28]: I went with the WildFly names. I looked at the OpenSSL wiki https://wiki.openssl.org/index.php/TLS1.3 and they use "ciphers" for specifying filters for <= 1.2 and "ciphersuites" for 1.3.
The difference is not just in the protocols they apply to, but in the way they affect the available ciphersuites. The filter allows selecting whole familiies. The names select specific ciphersuites. And unfortunately there is no overlap between the two.
I like your proposal, but I don't like hardcoding the version in the attribute name...

**Review by pruivo — APPROVED** [review]

**Comment by ryanemerson** [comment]
@tristantarrant `org.infinispan.server.configuration.ServerConfigurationSerializerTest` is failing:

```
org.infinispan.commons.CacheConfigurationException: org.infinispan.commons.configuration.io.ConfigurationReaderException: Unexpected attribute 'enabled-ciphersuites13' encountered[311,1]
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:163)
	at org.infinispan.server.configuration.ServerConfigurationSerializerTest.testConfigurationSerialization(ServerConfigurationSerializerTest.java:91)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:398)
	at org.apache.maven.surefire.junitcore.pc.InvokerStrategy.schedule(InvokerStrategy.java:54)
	at org.apache.maven.surefire.junitcore.pc.Scheduler.schedule(Scheduler.java:355)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:398)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.infinispan.commons.configuration.io.ConfigurationReaderException: Unexpected attribute 'enabled-ciphersuites13' encountered[311,1]
	at org.infinispan.configuration.parsing.ParseUtils.unexpectedAttribute(ParseUtils.java:64)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseSSLEngine(ServerConfigurationParser.java:1024)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseSSL(ServerConfigurationParser.java:1001)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseServerIdentities(ServerConfigurationParser.java:978)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseSecurityRealm(ServerConfigurationParser.java:451)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseSecurityRealms(ServerConfigurationParser.java:415)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseSecurity(ServerConfigurationParser.java:308)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseServerElements(ServerConfigurationParser.java:124)
	at org.infinispan.server.configuration.ServerConfigurationParser.readElement(ServerConfigurationParser.java:104)
	at org.infinispan.configuration.parsing.ParserRegistry.parseElement(ParserRegistry.java:209)
	at org.infinispan.configuration.parsing.ConfigurationBuilderHolder.handleAnyElement(ConfigurationBuilderHolder.java:139)
	at org.infinispan.commons.configuration.io.AbstractConfigurationReader.handleAny(AbstractConfigurationReader.java:57)
	at org.infinispan.configuration.parsing.Parser.readElement(Parser.java:99)
	at org.infinispan.configuration.parsing.ParserRegistry.parseElement(ParserRegistry.java:209)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:187)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:175)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:157)
```

**Comment by tristantarrant** [comment]
Using `enable-ciphersuites-tls13` is much better.

**Comment by ryanemerson** [comment]
Thanks @tristantarrant

**Pull request merged**